From 2e33976a1dc87b5d211159bfb4d5447404f2ed08 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:56 -0500
Subject: [drm] radeon: don't reset sdma on CIK init

Message-id: <1418852380-13061-46-git-send-email-rclark@redhat.com>
Patchwork-id: 102200
O-Subject: [RHEL7 drm 45/89] drm/radeon: don't reset sdma on CIK init
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit 7d102d16c64a0cd8dd98f9d81502292f04643360

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Thu Sep 18 10:18:43 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:58 2014 -0500

    drm/radeon: don't reset sdma on CIK init

    commit 799028d5d85384cce140323be633c8d5f079193f upstream.

    Otherwise we may lose the DMA golden settings which can
    lead to hangs, etc.

    Reviewed-by: Christian Koenig <christian.koenig@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/cik_sdma.c b/drivers/gpu/drm/radeon/cik_sdma.c
index 8e9d0f1..72bff72 100644
--- a/drivers/gpu/drm/radeon/cik_sdma.c
+++ b/drivers/gpu/drm/radeon/cik_sdma.c
@@ -459,13 +459,6 @@ int cik_sdma_resume(struct radeon_device *rdev)
 {
  int r;
 
- /* Reset dma */
- WREG32(SRBM_SOFT_RESET, SOFT_RESET_SDMA | SOFT_RESET_SDMA1);
- RREG32(SRBM_SOFT_RESET);
- udelay(50);
- WREG32(SRBM_SOFT_RESET, 0);
- RREG32(SRBM_SOFT_RESET);
-
  r = cik_sdma_load_microcode(rdev);
  if (r)
   return r;
-- 
1.7.1